content/v2.0/reference/syntax/annotated-csv/extended.md

```diff
@@ -0,0 +1,360 @@
+each column represents.
+To explicitly define a column as a **field** of a specific data type, use the field
+type in the annotation.
+
+| Data type                     | Resulting line protocol                 |
+|:----------                    |:-----------------------                 |
+| [measurement](#measurement)   | Column is the **measurement**           |
+| [tag](#tag)                   | Column is a **tag**                     |
+| [dateTime](#datetime)         | Column is the **timestamp**             |
+| [field](#field)               | Column is a **field**                   |
+| [ignored](#ignored)           | Column is ignored                       |
+| [string](#string)             | Column is a **string field**            |
+| [double](#double)             | Column is a **float field**             |
+| [long](#long)                 | Column is an **integer field**          |
+| [unsignedLong](#unsignedlong) | Column is an **unsigned integer field** |
+| [boolean](#boolean)           | Column is a **boolean field**           |
+
+#### measurement
+Indicates the column is the **measurement**.
+
+#### tag
+Indicates the column is a **tag**.
+The **column label** is the **tag key**.
+The **column value** is the **tag value**.
+
+#### dateTime
+Indicates the column is the **timestamp**.
+`time` is as an alias for `dateTime`.
+If the [timestamp format](#supported-timestamp-formats) includes a time zone,
+the parsed timestamp respects the time zone.
+By default, all timestamps are UTC.
+You can also use the [`#timezone` annotation](#timezone) to adjust timestamps to
+a specific time zone.
+
+{{% note %}}
+There can only be **one** `dateTime` column.
+{{% /note %}}
+
+The `influx write` command converts timestamps to [Unix timestamps](/v2.0/reference/glossary/#unix-timestamp).
+Append the timestamp format to the `dateTime` datatype with (`:`).
+
+```csv
+#datatype dateTime:RFC3339
+#datatype dateTime:RFC3339Nano
+#datatype dateTime:number
+#datatype dateTime:2006-01-02
+```
+
+##### Supported timestamp formats
+
+| Timestamp format | Description       | Example                          |
+|:---------------- |:-----------       |:-------                          |
+| **RFC3339**      | RFC3339 timestamp | `2020-01-01T00:00:00Z`           |
+| **RFC3339Nano**  | RFC3339 timestamp | `2020-01-01T00:00:00.000000000Z` |
+| **number**       | Unix timestamp    | `1577836800000000000`            |
+
+{{% note %}}
+If using the `number` timestamp format and timestamps are **not nanosecond Unix timestamps**,
+use the [`--precision` flag](/v2.0/reference/cli/influx/write/#flags) with the
+`influx write` command to specify the timestamp precision.
+{{% /note %}}
+
+##### Custom timestamp formats
+To specify a custom timestamp format, use timestamp formats as described in the
+[Go time package](https://golang.org/pkg/time).
+For example: `2020-01-02`.
+
+#### field
+Indicates the column is a **field** and auto-detects the field type.
```

Comment:
The current implementation of `field` field type does not detect the field type. It copies the field value as-is to line protocol. Maybe that we should better skip this being documented, we generally prefer specific field types.

Reply:
I think I'll leave it in, just because the functionality does exist. But I added a note recommending to specify the field type.